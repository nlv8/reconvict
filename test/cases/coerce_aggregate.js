'use strict';

exports.conf = {
  nested: {
    data: {
      format: 'map',
      default: {},
      arg: 'data'
    }
  }
}

exports.argv = '--data key=val --data foo=bar'

exports.formats = {
  map: {
    validate: function(value) {
      if (typeof value !== 'object') {
        throw new Error('must be a map of key/value pairs');
      }
    },
    coerce: function(value, config, path) {
      const accum = config.get(path);
      value.split(',').forEach(function(pair) {
        const [k, v] = pair.split('=');
        accum[k] = v;
      })
      return accum;
    }
  }
}
