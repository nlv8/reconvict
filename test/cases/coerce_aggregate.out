{
  "nested": {
    "data": {
      "foo": "bar",
      "key": "val"
    }
  }
}
